plugin/src/main/java/org/opensearch/ml/action/controller/GetModelControllerTransportAction.java

```diff
@@ -0,0 +1,150 @@
+import org.opensearch.commons.authuser.User;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.index.IndexNotFoundException;
+import org.opensearch.ml.common.MLModel;
+import org.opensearch.ml.common.controller.MLModelController;
+import org.opensearch.ml.common.transport.controller.MLModelControllerGetAction;
+import org.opensearch.ml.common.transport.controller.MLModelControllerGetRequest;
+import org.opensearch.ml.common.transport.controller.MLModelControllerGetResponse;
+import org.opensearch.ml.helper.ModelAccessControlHelper;
+import org.opensearch.ml.model.MLModelManager;
+import org.opensearch.ml.utils.RestActionUtils;
+import org.opensearch.search.fetch.subphase.FetchSourceContext;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+import lombok.AccessLevel;
+import lombok.experimental.FieldDefaults;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+@FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+public class GetModelControllerTransportAction extends HandledTransportAction<ActionRequest, MLModelControllerGetResponse> {
+    Client client;
+    NamedXContentRegistry xContentRegistry;
+    ClusterService clusterService;
+    MLModelManager mlModelManager;
+    ModelAccessControlHelper modelAccessControlHelper;
+
+    @Inject
+    public GetModelControllerTransportAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        Client client,
+        NamedXContentRegistry xContentRegistry,
+        ClusterService clusterService,
+        MLModelManager mlModelManager,
+        ModelAccessControlHelper modelAccessControlHelper
+    ) {
+        super(MLModelControllerGetAction.NAME, transportService, actionFilters, MLModelControllerGetRequest::new);
+        this.client = client;
+        this.xContentRegistry = xContentRegistry;
+        this.clusterService = clusterService;
+        this.mlModelManager = mlModelManager;
+        this.modelAccessControlHelper = modelAccessControlHelper;
+    }
+
+    @Override
+    protected void doExecute(Task task, ActionRequest request, ActionListener<MLModelControllerGetResponse> actionListener) {
+        MLModelControllerGetRequest modelControllerGetRequest = MLModelControllerGetRequest.fromActionRequest(request);
+        String modelId = modelControllerGetRequest.getModelId();
+        FetchSourceContext fetchSourceContext = getFetchSourceContext(modelControllerGetRequest.isReturnContent());
+        GetRequest getRequest = new GetRequest(ML_MODEL_CONTROLLER_INDEX).id(modelId).fetchSourceContext(fetchSourceContext);
+        User user = RestActionUtils.getUserContext(client);
+        String[] excludes = new String[] { MLModel.MODEL_CONTENT_FIELD, MLModel.OLD_MODEL_CONTENT_FIELD };
```

Comment:
minor: keep this creating variable line close to the line where it's being used : line 87

Reply:
Got it

Reply:
It seems all the code keeps this line out of the try catch block, will keep aligned to this.